x-pack/plugins/security_solution/public/resolver/data_access_layer/mocks/no_resolver_data_returned.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ResolverRelatedEvents,
+  ResolverTree,
+  ResolverEntityIndex,
+} from '../../../../common/endpoint/types';
+import { DataAccessLayer } from '../../types';
+
+interface Metadata {
+  /**
+   * The `_id` of the document being analyzed.
+   */
+  databaseDocumentID: string;
+  /**
+   * A record of entityIDs to be used in tests assertions.
+   */
+  entityIDs: {};
+}
+
+/**
+ * A simple mock dataAccessLayer that rejects on any data requests.
+ */
+export function noResolverDataReturned(): {
+  dataAccessLayer: DataAccessLayer;
+  metadata: Metadata;
+} {
+  const metadata: Metadata = {
+    databaseDocumentID: '_noDataID',
+    entityIDs: {},
+  };
+  return {
+    metadata,
+    dataAccessLayer: {
+      /**
+       * Fetch related events for an entity ID
+       */
+      relatedEvents(entityID: string): Promise<ResolverRelatedEvents> {
+        return Promise.reject(Error('Events are unable to be found'));
```

Comment:
Curious about this. How closely does this mimic the behavior of the actual data access layer? I was kinda under the impression that we don't ever expect error responses from the HTTP server.

Reply:
Yea, I've updated it to follow expected patterns.